### ISSUE TITLE
VR

### ISSUE THREAD

**Issue by TheTrueShadow** [nonamecrackers2/crackers-wither-storm-mod#393]
Add VR support

**Comment by ghost** [comment]
yes, and play it a 3 am, max volume.

**Comment by ghost** [comment]
All jokes aside i love this idea and i want to do this with freinds.

**Comment by watersofnazareth** [comment]
It does work in VR

**Comment by nonamecrackers2** [comment]
It’s mostly compatible except for the distant renderer. I’ll mark this as a big but it’s quite low priority.

**Comment by V0calgiant12** [comment]
do you think that now with Vivecraft being rewritten semi-recently, do you think that it would be possible to fix the distant render? I mean, i dont know that much about the code but to me it seems like it might be fixable.

**Comment by nonamecrackers2** [comment]
Probably not, I have an idea on why it's not working properly and I don't believe it would be related to that.

**Comment by V0calgiant12** [comment]
oh, I thought that it was being caused by trying to render both screens of the eyes from 1 point, offsetting the distant render on the eye screens. Kind of like a triangle, there's one point at the top of the triangle that is also the midpoint. And both of the other 2 points would, most likely, see it from 2 different points of view. Here, I made a picture if you didnt understand any of that

TL;DR: look at this picture I made that I think might be the problem
![VRWS](https://user-images.githubusercontent.com/109500471/209457434-23267197-261b-4ab8-a2ca-de368b5e20ec.png)

**Comment by nonamecrackers2** [comment]
I'm making guesses here but I'm 99% sure this is why:

When rendering vertices, a projection matrix transforms these vertices and also defines the maximum and minimum distance these vertices can be rendered (called the "near plane" and "far plane"). Minecraft's default far plane distance is quite short, and to allow the Wither Storm to be viewed from much farther away we need to create a new projection matrix with an extended far plane distance. This is a key component in what allows the Wither Storm to be viewed from up to 10K blocks away.

Vivecraft I believe modifies the projection matrix to make the world rotate and move with your head movements. However, when it gets to the distant renderer, CWSM effectively uses a different projection matrix which does not have the movements and rotations of your head applied. This is why the Wither Storm does not render properly on the distant renderer. Your diagrams do definitely get part of the picture however.

**Comment by V0calgiant12** [comment]
ahh, that makes a lot of sense when put like that and I can see how that takes effect now (as I have experienced a distant render wither storm spinning around my head myself)
Maybe if you get in contact with the people who like just rewrote all of the code for ViveCraft, maybe they could help you get it fixed? Because Im assuming the reason why you cant fix it atm is because you dont know what the ViveCraft code looks like for that, correct? if so, maybe they could provide what is needed to fix it.

**Comment by nonamecrackers2** [comment]
Possibly yeah

**Comment by nonamecrackers2** [comment]
Fixed in 2.2/3.2
